common-npm-packages/azure-arm-rest-v2/azure-graph.ts

```diff
@@ -55,28 +58,34 @@ export class ServicePrincipals {
         httpRequest.method = 'GET';
         httpRequest.headers = this.client.setCustomHeaders(options);
 
-        var filterQuery = util.format("appId eq '%s'", this.client.getCredentials().getClientId());
-        httpRequest.uri = this.client.getRequestUri("{tenantId}/servicePrincipals",
+        const clientCredentials = this.client.getCredentials();
+        const useMSAL = clientCredentials.getUseMSAL();
+        tl.debug(`MSAL - Graph - GetServicePrincipal - useMSAL = ${useMSAL}`);
+        const requestUriFormat = (useMSAL ? "v1.0/" : "") + "{tenantId}/servicePrincipals";
```

Comment:
Is this prefix required just for service principals API? Should not we prefix all requests in `AzureServiceClientBase.getRequestUriForBaseUri` based on `credentials.getUseMsal`?